framework/src/base/FEProblemBase.C

```diff
@@ -364,6 +364,8 @@ void FEProblemBase::initialSetup()
   if (_app.isRecovering() && (_app.isUltimateMaster() || _force_restart))
     _resurrector->setRestartFile(_app.getRecoverFileBase());
 
+  _mesh.initialSetup();
```

Comment:
Well that may not be the right place. On recover for instance you might need information from the recovery before running the `initialSetup()` method. What are you planning on doing in your specific case?